nest-tax-backend/src/tasks/tasks.service.ts

```diff
@@ -176,4 +181,92 @@ export class TasksService {
       config.REPORTING_RECIPIENT_EMAIL,
     )
   }
+
+  @Cron(CronExpression.EVERY_DAY_AT_1AM)
+  @HandleErrors('Cron Error')
+  async sendUnpaidTaxReminders() {
+    if (process.env.FEATURE_TOGGLE_REMINDER_UNPAID_TAX !== 'true') {
+      return
+    }
+    const TWENTY_DAYS_AGO = dayjs().subtract(20, 'day').toDate() // 20 days ago
+    const taxes = await this.prismaService.tax.findMany({
+      select: {
+        id: true,
+        year: true,
+        taxPayer: {
+          select: {
+            birthNumber: true,
+          },
+        },
+      },
+      where: {
+        bloomreachUnpaidTaxReminderSent: false,
+        taxPayments: {
+          none: {
+            status: PaymentStatus.SUCCESS,
+          },
+        },
+        OR: [
+          {
+            deliveryMethod: DeliveryMethodNamed.CITY_ACCOUNT,
+            createdAt: {
+              lte: TWENTY_DAYS_AGO,
+            },
+          },
+          {
+            deliveryMethod: {
+              not: DeliveryMethodNamed.CITY_ACCOUNT,
+            },
+            dateTaxRuling: {
+              lte: TWENTY_DAYS_AGO,
+            },
+          },
+          {
+            deliveryMethod: null,
+            dateTaxRuling: {
+              lte: TWENTY_DAYS_AGO,
+            },
+          },
+        ],
+      },
+    })
+
+    if (taxes.length === 0) return
```

Comment:
Please wrap this return into brackets, as it makes it less error prone to future enhancements in this code block (for instance some log might be needed) and it makes it more readable in my opinion.
```suggestion
    if (taxes.length === 0) {
      return
    }
```